nx/lib/nx/binary_backend.ex

```diff
@@ -829,6 +829,26 @@ defmodule Nx.BinaryBackend do
   @impl true
   def conjugate(out, tensor), do: element_wise_unary_op(out, tensor, &Complex.conjugate/1)
 
+  @impl true
+  def real(out, tensor) do
+    real_fn = fn
+      %Complex{re: re} -> re
+      num -> num
+    end
+
+    element_wise_unary_op(out, tensor, real_fn)
+  end
+
+  @impl true
+  def imag(out, tensor) do
+    imag_fn = fn
+      %Complex{im: im} -> im
+      _ -> 0.0
+    end
+
+    element_wise_unary_op(out, tensor, imag_fn)
```

Comment:
We can specialize this too. For complex numbers, we get the second part of the binary. For all others, we can do a `:binary.duplicate(number_to_binary(0, out.type), size)`.

Reply:
I'll rework `real` and `imag` here :)